### PR TITLE
Revert Boost BigInt workaround

### PR DESCRIPTION
**Pull request by darkdh**
Resolves https://github.com/brave/brave-browser/issues/24920
Resolves https://github.com/brave/brave-browser/issues/21453 which should no longer be valid.

## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### CODE REVIEW

**Review by bridiver — APPROVED** [review]

File: components/brave_wallet/common/brave_wallet_types.h [hunk 1]
```diff
@@ -11,29 +11,25 @@
...
 
 #include "base/values.h"
-#include "boost/multiprecision/cpp_int.hpp"
 #include "third_party/abseil-cpp/absl/types/optional.h"
 
 namespace brave_wallet {
 
-typedef boost::multiprecision::uint256_t uint256_t;
-typedef boost::multiprecision::int256_t int256_t;
+typedef unsigned _BitInt(256) uint256_t;
```
Comment (supermassive) [inline components/brave_wallet/common/brave_wallet_types.h:18]: Can we use `using` form here? It is a preferred style through chromium.

File: components/brave_wallet/common/brave_wallet_types.h [hunk 2]
```diff
@@ -11,29 +11,25 @@
...
 // -(2^255 -1)
-constexpr int256_t kMin256BitInt = std::numeric_limits<int256_t>::min() >> 1;
+constexpr int256_t kMin256BitInt = std::numeric_limits<int256_t>::min();
 
 // 2^128 - 1
-constexpr int128_t kMax128BitInt = std::numeric_limits<int128_t>::max() >> 1;
+constexpr int128_t kMax128BitInt = std::numeric_limits<int128_t>::max();
 // -(2^128 -1)
-constexpr int128_t kMin128BitInt = std::numeric_limits<int128_t>::min() >> 1;
+constexpr int128_t kMin128BitInt = std::numeric_limits<int128_t>::min();
```
Comment (supermassive) [inline components/brave_wallet/common/brave_wallet_types.h:32]: I believe some comments above are incorrect now :)
Actually do we need these constants? The only place references them is Base10ValueToInt256 which should be also be fixed now as it compares int256_t with it's true min/max.

Comment (darkdh) [reply components/brave_wallet/common/brave_wallet_types.h:32]: Those comments are correct, it was shifted not minus

Comment (darkdh) [reply components/brave_wallet/common/brave_wallet_types.h:32]: And of course we can use  std::numeric_limits<*> directly.

File: components/brave_wallet/common/brave_wallet_types.h (hunk 1 above)
Comment (darkdh) [reply components/brave_wallet/common/brave_wallet_types.h:18]: Sure, I don't see why not and using works in template.

File: components/brave_wallet/common/brave_wallet_types.h (hunk 2 above)
Comment (darkdh) [reply components/brave_wallet/common/brave_wallet_types.h:32]: Oh I see what you mean by incorrect now, that 128 bit comment was added by different person and it is indeed wrong.

File: components/brave_wallet/common/brave_wallet_types.h (hunk 1 above)
Comment (darkdh) [reply components/brave_wallet/common/brave_wallet_types.h]: fixed and force pushed.

File: components/brave_wallet/common/brave_wallet_types.h (hunk 2 above)
Comment (darkdh) [reply components/brave_wallet/common/brave_wallet_types.h:32]: comments fixed and force pushed.

**Comment by supermassive** [comment]
LGTM

**Review by supermassive — APPROVED** [review]

**Pull request merged**